Assets/Editor/E07_Data.cs

```diff
@@ -0,0 +1,151 @@
+                    Debug.Assert(s_Asset != null);
+                }
+                return s_Asset;
+            }
+        }
+
+        List<int> m_ExpandedGOs;
+
+        public void OnEnable()
+        {
+            if (m_ExpandedGOs == null)
+                m_ExpandedGOs = new List<int>();
+            
+            Scene scene = SceneManager.GetSceneAt(0);
+
+            var scrollView = new ScrollView();
+            scrollView.StretchToParentSize();
+
+            if (scene.IsValid())
+            {
+                GameObject[] gameObjects = scene.GetRootGameObjects();
+                RecursiveSetUp(gameObjects, scrollView.contentView);
+            }
+
+            this.GetRootVisualContainer().AddChild(scrollView);
+            this.GetRootVisualContainer().AddStyleSheetPath("TreeView");
+        }
+
+        static void UpdateName(VisualElement treeViewItem, Object data)
+        {
+            if (data == null)
+                return;
+
+            treeViewItem.Q<Label>().text = data.name;
+        }
+
+        public void UpdateChildren(VisualElement treeViewItem, Object data)
+        {            
+            Transform transform = data as Transform;           
+
+            // Might be deleted
+            if (transform == null)
+                return;
+
+            List<GameObject> gos = new List<GameObject>();
+            for (int i = 0; i < transform.childCount; i++)
+            {
+                gos.Add(transform.GetChild(i).gameObject);
+            }
+            Toggle toggle = treeViewItem.Q<Toggle>();
+			toggle.enabled = gos.Count > 0;
+			toggle.on = toggle.enabled && m_ExpandedGOs.Contains(transform.gameObject.GetInstanceID());
+
+            if (m_ExpandedGOs.Contains(transform.gameObject.GetInstanceID()))
+            {
+                RecursiveSetUp(gos, treeViewItem.Q<VisualContainer>("ChildrenContainer"));    
+            }
+        }
+
+        public void RecursiveSetUp(IEnumerable<GameObject> gameObjects, VisualContainer container)
+        {
+            var currentGOs = new List<GameObject>();
+            container.ToList().ForEach((e) => {
+                GameObject go = e.data as GameObject;
+
+                // remove gone objects
+                if (go == null || !gameObjects.Contains(go))
+                {
+                    container.RemoveChild(e);
+                }
+                // or-remember those that already have an item
+                else
+                {
+                    currentGOs.Add(go);
+                }
+            });
+
+
+            foreach(GameObject go in gameObjects.Except(currentGOs))
+            {
+                VisualContainer added = asset.CloneTree();
+                added.data = go;
+
+                UpdateName(added, go);
+                added.RegisterWatch(go, UpdateName);
+				UpdateChildren(added, go.transform);
+				added.RegisterWatch(go.transform, UpdateChildren);
+				
```

Comment:
As discussed, we need a lambda here because the delegate does not send the VE with it.
We should change all VisualElement delegates to send the 'this' as context.